source/adapters/level_zero/context.cpp

```diff
@@ -656,9 +657,17 @@ ur_result_t ur_context_handle_t_::getAvailableCommandList(
   // for this queue.
   if (Queue->hasOpenCommandList(UseCopyEngine)) {
     if (AllowBatching) {
-      CommandList = CommandBatch.OpenCommandList;
-      UR_CALL(Queue->insertStartBarrierIfDiscardEventsMode(CommandList));
-      return UR_RESULT_SUCCESS;
+      bool batchingAllowed = true;
```

Comment:
it seems you've added `uint32_t NumEventsInWaitList, const ur_event_handle_t *EventWaitList` arguments just to optionally change the value of `AllowBatching`, which is also an argument to this function. Would it make sense to move checking 'allow batching' to the caller?